Makefile

```diff
@@ -33,13 +35,8 @@ PROTOC_ARTIFACTS := pkg/pbflow
 EXCLUDE_COVERAGE_FILES="(/cmd/)|(bpf_bpfe)|(/examples/)|(/pkg/pbflow/)"
 
 # Image building tool (docker / podman)
-ifndef OCI_BIN
-	ifeq (,$(shell which podman 2>/dev/null))
-	OCI_BIN=docker
-	else
-	OCI_BIN=podman
-	endif
-endif
+OCI_BIN_PATH = $(shell which podman  || which docker)
```

Comment:
*   If podman doesn't exist, for example, it will display an error message.  This should be suppressed. 
*   It should also check that one of podman or docker exists and quit if that is not the case.
*   Use := instead of =.  
      
Suggest:  
```
OCI_BIN_PATH := $(shell which podman 2> /dev/null || which docker 2> /dev/null)
ifeq ($(OCI_BIN_PATH),)
$(error Requires podman or docker in PATH)
endif
```

Reply:
Yes, I relied on [FLP Makefile](https://github.com/netobserv/flowlogs-pipeline/blob/main/Makefile#L17-L18) for such changes. Let's all agree on how to manage that and apply it everywhere !

@jotak @OlivierCazade  @msherif1234

Reply:
How about a slight variation:

```
OCI_BIN_PATH ?= $(shell which podman 2> /dev/null || which docker 2> /dev/null)
ifeq ($(OCI_BIN_PATH),)
$(error Requires OCI-compliant CLI such as podman or docker in PATH)
endif
```

`?=` allows for overriding it, e.g. using a different installation path or another OCI compliant tool

Reply:
The only problem is that in other parts of the Makefile, it generally uses OCI\_BIN, which is the basename of OCI\_BIN\_PATH (either 'podman' or 'docker') instead of OCI\_BIN\_PATH which has the absolute path.  Therefore, it's possible if you override OCI\_BIN\_PATH, OCI\_BIN could fail or even use a different version depending on PATH.  
  
You could change all OCI\_BIN usage to OCI\_BIN\_PATH.  There is at least one case where it's using OCI\_BIN to determine if you are running docker or podman.

Reply:
your suggestions doesn't work with `docker.io/library/golang:1.19` image + it's common to simply use `$(shell which podman  || which docker)` without further checks

I'll keep this as is for now